### PR TITLE
Render components outside of ASPNETCORE

### LINKED ISSUE #28980: Render Razor components outside of ASP.NET Core
The content doesn't neatly fit into existing Blazor coverage because outside of dedicated SPA docs everything else is specific to Blazor apps. It looks like this can be a separate doc under the *Components* node for now with final location TBD, perhaps when Blazor United coverage is worked out around Pre5.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #28980

Notes:

* The content doesn't really fit into Blazor-specific docs, so I create an article for it in the *Components* node. I also didn't want to try and tack this on to the *Components* node overview article because that article is already *toooo looong* 😩. I'm actually going to be looking for ways to shorten it further, not make it longer. We can move the content of this new article later easily enough. The nice thing here is to have the content work done (or mostly done).
* Code tested working locally here.
* I have a tracking item in my 8.0 tracking issue to strip out the `--prerelease` options at RTM.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/components/render-components-outside-of-aspnetcore.md](https://github.com/dotnet/AspNetCore.Docs/blob/7e92944b73d166387a62910347813a1120e722d9/aspnetcore/blazor/components/render-components-outside-of-aspnetcore.md) | [aspnetcore/blazor/components/render-components-outside-of-aspnetcore](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?branch=pr-en-us-28981) |

### CODE REVIEW

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @guardrex!
I think we want to squeeze the `templating engine` wording somewhere in the list of scenarios this can be used in, as that's the context the ask would periodically come up in the past. That will help with SEO in the future.

**Comment by guardrex** [comment]
@mkArtakMSFT ... I made the third opening sentence ...

> This is convenient for scenarios where you want to generate HTML fragments, such as for generating email content, generating static site content, or for building a content templating engine.

**Comment by guardrex** [comment]
Sorry Dan ... I did mean to hit you up for review on this before merging. Let me know if you spot anything, and I'll get a patch PR up. I'm just trying to 🏃🏃🏃 and get this whole batch of PRs merged asap to unblock other work on these articles.

**Pull request merged**
